### ISSUE TITLE
[Card] Additional sizes and names

### ISSUE THREAD

**Issue by dianatran18** [carbon-design-system/carbon-addons-iot-react#3283]
<ins style="box-sizing: border-box; margin-top: 0px !important; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><h3 class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; font-size: 1.25em; font-weight: 600; line-height: 1.25; margin-left: 16px;">What package is this for?</h3></ins><ul class="contains-task-list added rich-diff-level-zero" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 16px; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; background: var(--color-success-subtle); text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><li class="task-list-item rich-diff-level-one" style="box-sizing: border-box; list-style-type: none; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;"><input type="checkbox" id="" disabled="" class="task-list-item-checkbox" checked="" style="box-sizing: border-box; font: inherit; margin: 0px 0.2em 0.25em -1.6em; overflow: visible; padding: 0px; vertical-align: middle; transition: color 80ms cubic-bezier(0.33, 1, 0.68, 1) 0s, background-color, box-shadow, border-color;"><label style="box-sizing: border-box; font-weight: 400;"></label><span> </span>React</li><li class="task-list-item rich-diff-level-one" style="box-sizing: border-box; list-style-type: none; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;"><input type="checkbox" id="" disabled="" class="task-list-item-checkbox" style="box-sizing: border-box; font: inherit; margin: 0px 0.2em 0.25em -1.6em; overflow: visible; padding: 0px; vertical-align: middle; transition: color 80ms cubic-bezier(0.33, 1, 0.68, 1) 0s, background-color, box-shadow, border-color;"><label style="box-sizing: border-box; font-weight: 400;"></label><span> </span>Angular</li></ul><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><h3 class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.25em; font-weight: 600; line-height: 1.25; margin-left: 16px;">Summary</h3></ins><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;"><a href="https://pages.github.ibm.com/ai-applications/design/components/card/usage/" rel="nofollow" class="rich-diff-level-one" style="box-sizing: border-box; background-color: transparent; color: var(--color-accent-fg); text-decoration: none; transition: color 80ms cubic-bezier(0.33, 1, 0.68, 1) 0s, background-color, box-shadow, border-color;">PAL card guidance</a></p></ins><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;">In the PAL, the card component has 5 card sizes and a different naming structure than what is in React. This was due to the card existing in code before with no documentation on the site, and then the dashboard guild working on cards later on.</p></ins><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;">I've updated the guidance to align on the sizes available in React, but this issue is to also include 2 additional card sizes, and to update the naming convention we have for the card widths and heights.</p></ins><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;">After talking to<span> </span><a class="user-mention notranslate rich-diff-level-one" data-hovercard-type="user" data-hovercard-url="/users/davidicus/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/davidicus" style="box-sizing: border-box; background-color: transparent; color: var(--color-fg-default); text-decoration: none; transition: color 80ms cubic-bezier(0.33, 1, 0.68, 1) 0s, background-color, box-shadow, border-color; font-weight: 600; white-space: nowrap;">@davidicus</a><span> </span>about it, we won't be replacing the existing names for those using it, but adding additional names to the card sizes.</p></ins><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;"><a target="_blank" rel="noopener noreferrer" href="https://user-images.githubusercontent.com/43549567/153619470-c76e0346-38d6-4f2d-a333-a996c34e5517.png" class="rich-diff-level-one" style="box-sizing: border-box; background-color: transparent; color: var(--color-accent-fg); text-decoration: none; transition: color 80ms cubic-bezier(0.33, 1, 0.68, 1) 0s, background-color, box-shadow, border-color;"><img src="https://user-images.githubusercontent.com/43549567/153619470-c76e0346-38d6-4f2d-a333-a996c34e5517.png" alt="card-sizes-2" style="box-sizing: content-box; border-style: none; max-width: 100%; background-color: var(--color-canvas-default);"></a></p></ins><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;">Card widths:</p></ins><ul class="added rich-diff-level-zero" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 16px; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; background: var(--color-success-subtle); text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Extra small: 4 columns</li><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Small: 6 columns (new size)</li><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Medium: 8 columns</li><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Large: 12 columns (new size)</li><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Extra large: 16 columns</li></ul><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;">Card heights:</p></ins><ul class="added rich-diff-level-zero" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 16px; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; background: var(--color-success-subtle); text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Short: 144 px</li><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Regular: 304 px</li><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Tall: 464 px (new size)</li><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Extra tall: 624 px</li></ul><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;">Based on what I see in the [storybook], the 1:1 mapping for these new names would be:</p></ins><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);">

Existing name | New name
-- | --
SMALL | Extra small, Short
SMALLWIDE | Medium, Short
SMALLFULL | Extra large, Short
MEDIUMTHIN | Extra small, Regular
MEDIUM | Medium, Regular
MEDIUMWIDE | Extra large, Regular
LARGETHIN | Extra small, Extra tall
LARGE | Medium, Extra tall
LARGEWIDE | Extra large, Extra tall

</ins><ins style="box-sizing: border-box; margin-bottom: 0px !important; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;">Would it make sense to separate out the width and height sizes for more flexibility?</p></ins>What package is this for?
 React
 Angular
Summary
[PAL card guidance](https://pages.github.ibm.com/ai-applications/design/components/card/usage/)

In the PAL, the card component has 5 card sizes and a different naming structure than what is in React. This was due to the card existing in code before with no documentation on the site, and then the dashboard guild working on cards later on.

I've updated the guidance to align on the sizes available in React, but this issue is to also include 2 additional card sizes, and to update the naming convention we have for the card widths and heights.

After talking to @davidicus about it, we won't be replacing the existing names for those using it, but adding additional names to the card sizes.

card-sizes-2

Card widths:

Extra small: 4 columns
Small: 6 columns (new size)
Medium: 8 columns
Large: 12 columns (new size)
Extra large: 16 columns
Card heights:

Short: 144 px
Regular: 304 px
Tall: 464 px (new size)
Extra tall: 624 px
Based on what I see in the [storybook], the 1:1 mapping for these new names would be:

Existing name	New name
SMALL	Extra small, Short
SMALLWIDE	Medium, Short
SMALLFULL	Extra large, Short
MEDIUMTHIN	Extra small, Regular
MEDIUM	Medium, Regular
MEDIUMWIDE	Extra large, Regular
LARGETHIN	Extra small, Extra tall
LARGE	Medium, Extra tall
LARGEWIDE	Extra large, Extra tall
Would it make sense to separate out the width and height sizes for more flexibility?

**Comment by JordanWSmith15** [comment]
@davidicus @dianatran18 "we won't be replacing the existing names for those using it, but adding additional names to the card sizes."   I guess that would be because chaning the names would break a lot of contracts downstream. Would this mean that we expose duplicative size options, though? It seems like we are only adding more potentialy for confusion, since none of these names are ever referenced in the designs handed over to development anyway (that I know of)

**Comment by JordanWSmith15** [comment]
Blocks https://jsw.ibm.com/browse/GRAPHITE-48767

**Comment by davidicus** [comment]
@JordanWSmith15 @jessieyan yes. We can't remove the old names but we will take them out of the documentation which should be done along with this work. Instead just referencing the new "official" names.

**Comment by jessieyan** [comment]
@dianatran18 Hi Diana, would you please offer more information on the size dimensions? 

For example, in the existing design we have 6 breakpoints for size `SMALL`. 

 SMALL: {
    max: { w: 4, h: 1 },
    xl: { w: 4, h: 1 },
    lg: { w: 4, h: 1 },
    md: { w: 2, h: 1 },
    sm: { w: 2, h: 1 },
    xs: { w: 4, h: 1 },
  },

for breakpoint `max`, we have `width: 4 columns` and `height: 1 row`.


Do you have anything similar for the new names? Thank you!

**Comment by davidicus** [comment]
@dianatran18 Please help us with the breakpoint mapping. Also the image above does not take into account for 2 column cards. @JordanWSmith15 Correct me if I am  wrong, but this was an issue for Health the last time we updated our grid sizes and we had to revert some changes.

**Comment by JordanWSmith15** [comment]
@davidicus @jessieyan we should only be adding new sizes, not altering existing ones, so I don't expect that we'd effect any app teams downstream

**Comment by davidicus** [comment]
@jessieyan we will have to talk about the best way to support these new sizes without breaking existing dashboards. The small, for instance, is calling for 6 columns but, as you pointed out in your comment above, we have historically had 4 columns wide for Small cards.

**Comment by davidicus** [comment]
@JordanWSmith15  We can add the names, but from the image above in the issue description it shows a new mapping to the amount of grid rows. We can add the new names and then change new sizes in major release. Or we can provide a modified name to support the official grid size until next major.

**Comment by JordanWSmith15** [comment]
I don't like changing the names of the existing sizes much myself, as that definitely would effect downstream consumers. I am only interested in making the new sizes  available, particularly the 12 column width ones.

**Comment by dianatran18** [comment]
The confusing part is that "small" in the existing card naming convention is referring to the height of the card. In the new naming convention, "small" is referring to the width. 

<img width="1418" alt="Screen Shot 2022-06-09 at 11 08 43 PM" src="https://user-images.githubusercontent.com/43549567/172982594-32fff3b6-f6f9-49f8-aede-0566c3c88194.png">

Small, medium, large, etc. makes more sense as a width than it does a height. That follows what other components use, like [modals](https://carbondesignsystem.com/components/modal/style#sizes). The current naming wasn't consistent either. For example, the "SMALLWIDE" is 8 columns, but "MEDIUMWIDE" and "LARGEWIDE" are 16 columns. Short, regular, tall, etc. makes a lot more sense as heights. 

To account for the 2 column card size, it would be something more like this... I know teams have been using it already, but I still wouldn't surface that card size, because it can barely fit any content, and is not scalable. I just put "mini" for now, but I'll confirm with Robin on what we could call it. 

<img width="1390" alt="Screen Shot 2022-06-09 at 11 10 42 PM" src="https://user-images.githubusercontent.com/43549567/172982807-5e25ae7b-5bd8-4035-b305-5f3ab071388a.png">

**Comment by dianatran18** [comment]
Trying to shoehorn additional card sizes in an existing naming convention that is already confusing isn't a good solution. Is there any way we could create a fresh set of cards with the new naming convention and new card sizes in the major release, that are customizable so teams can put anything they want in them? I thought the cards now are not as flexible as we'd like anyways, so it seems like a good opportunity to correct the cards and build them to be more flexible.

**Comment by JordanWSmith15** [comment]
@dianatran18 Agree that we need to update this, but the there is a blocking dependency side and a technical debt side to this story that will need to be split out. Let me explain a bit:
For the blocking dependency, we have designers using the 12 column width sizes in designs despite the fact that they are not available in code (our typical issue), and that is causing application development teams to be dependent on us to deliver that 12 column width size. The app developers don't have a strong preference what the sizes are called... they will figure out the names. They really just need that size available to build the app to spec. 

On the technical debt side, we need to update ALL of the card size names. This will be a bit of work on the PAL side, but the main issue is that it exposes our consumers to a considerable amount of risk. If we make this change, and those teams update PAL, their application will break completely because their code is targeting the existing sizes. There is no way for us to update their code automatically to reference the new sizes. Someone from each app team will need to manually go through and change "Medium" to "Medium-Regular", etc. For @davidicus and I, knowingly delivering a breaking change of this kind downstream is a big deal because it's essentially our commitment to these teams that we'll never do such a thing. We could post a warning in the PAL React slack channel to give them a heads up, but that's the best we could do, and it would likely go unnoticed by a lot of impacted parties.  Even worse, the effort for a team like Tririga to upgrade their code, snapshots and automation might take a day or it might be huge... we can't tell how much pain we're inflicting, because we don't know how their code is structured.

So, we need to split out these two items into different deliveries.

**Comment by jessieyan** [comment]
Need to discuss with @davidicus how to get the new sized (E.g. Large 12 columns cards) as well as other combinations implemented and if we should add width prop and height prop to fulfill the width/height combinations.

@dianatran18 We still need  information on the dimensions of breakpoints for each card size.

**Comment by davidicus** [comment]
Hi @dianatran18, can we get your input for the breakpoints that Jessie is asking for?

**Comment by dianatran18** [comment]
Let me know if this works for the 12 column size only. 

@davidicus feel free to suggest a better name if these don't work.

SMALLTHICK: {
    max: { w: 12, h: 1 },
    xl: { w: 12, h: 1 },
    lg: { w: 12, h: 1 },
    md: { w: 8, h: 1 },
    sm: { w: 8, h: 1 },
    xs: { w: 4, h: 1 },
  },
MEDIUMTHICK: {
    max: { w: 12, h: 2 },
    xl: { w: 12, h: 2 },
    lg: { w: 12, h: 2 },
    md: { w: 8, h: 2 },
    sm: { w: 8, h: 2 },
    xs: { w: 4, h: 2 },
  },
LARGETHICK: {
    max: { w: 12, h: 4 },
    xl: { w: 12, h: 4 },
    lg: { w: 12, h: 4 },
    md: { w: 8, h: 4 },
    sm: { w: 8, h: 4 },
    xs: { w: 4, h: 4 },
  },

**Comment by jessieyan** [comment]
Hi Diana,

Thanks for the feedback. For all the new sizes showed up in the picture, do we need breakpoints for each one of them? Or could we have breakpoints just for the specific width(4, 6, 8, 12, 16) and height(1, 2, 3, 4) rather than size names.

Does SMALLTHICK, MEDIUMTHICK, LARGETHICK mean SMALLWIDE, MEDIUMWIDE and LARGEWIDE in the picture? 

![image](https://user-images.githubusercontent.com/24279794/177357497-92697076-2413-4e7e-a607-de935b2bc279.png)

**Comment by dianatran18** [comment]
The boxes highlighted in purple are the new 12 column sizes with the 3 heights. Those are the ones I listed out in the breakpoints above, and should be able to fit the needs @JordanWSmith15's team has. 

<img width="1523" alt="Screen Shot 2022-07-05 at 11 12 43 AM" src="https://user-images.githubusercontent.com/43549567/177360461-4669405a-9704-40cc-beef-2fc2f96d8e12.png">


As for the larger changes (adding additional sizes, updating the naming convention to the new structure), I think we'll need to break out that work separately. It will be super confusing to try and solve for both at once. @davidicus and I talked about potentially having a Card V2 with all of the updated names and sizes, and to control the width and the height separately, vs needing to control the widths and heights together with the current naming structure.
